nautobot_circuit_maintenance/tests/test_app_metrics.py

```diff
@@ -0,0 +1,48 @@
+"""Test cases for application metrics endpoint views."""
+from datetime import datetime, timedelta
+from django.test import TestCase
+from nautobot.circuits.models import Circuit, CircuitType, Provider, CircuitTermination
+from nautobot.dcim.models import Site
+
+from nautobot_circuit_maintenance.metrics_app import metric_circuit_operational
+from nautobot_circuit_maintenance.models import CircuitMaintenance, CircuitImpact
+
+
+class AppMetricTests(TestCase):
+    """Test cases for ensuring application metric endpoint is working properly."""
+
+    def setUp(self):
+        """Setup objects to run the test."""
+        self.provider = Provider.objects.create(name="Provider 1", slug="provider-1")
+        self.circuit_type = CircuitType.objects.create(name="Circuit Type 1", slug="circuit-type-1")
+        self.circuit = Circuit.objects.create(cid="Circuit 1", provider=self.provider, type=self.circuit_type)
+        self.circuit_2 = Circuit.objects.create(cid="Circuit 2", provider=self.provider, type=self.circuit_type)
+        self.circuit_3 = Circuit.objects.create(cid="Circuit 3", provider=self.provider, type=self.circuit_type)
+        self.site = Site.objects.create(name="Site 1", slug="site-1")
```

Comment:
Consider using different `type`, `provider`, and/or `site` values for each circuit so as to be able to test that the correct values are returned for each metric? Right now by using the same values across all circuits you're missing out on testing for some possible classes of bugs.